compiler/rustc_span/src/lib.rs

```diff
@@ -587,6 +587,12 @@ impl Span {
         matches!(outer_expn.kind, ExpnKind::Macro(..)) && outer_expn.collapse_debuginfo
     }
 
+    /// Returns `true` if `span` originates in a macros 2.0 expansion
```

Comment:
Why macros 2.0 specifically, why is it relevant to this error?

Reply:
Maybe I'm confused on nomenclature, the wrong help is emitted for proc macros only, with declarative macros the compiler puts the suggestion in the correct place. Aren't proc marcos and macros 2.0 the same thing essentially? If not this should probably just read `proc macro expansion` instead of `macros 2.0 expansion`

Reply:
>Aren't proc marcos and macros 2.0 the same thing essentially?

No, not at all.
Both declarative and procedural macros can use both old and macro 2.0 hygiene.

This situation is far from being unique, I suggest checking what other diagnostics do (`fn in_external_macro` or something like that).